core/trino-main/src/main/java/io/trino/operator/StreamingAggregationOperator.java

```diff
@@ -275,6 +275,9 @@ private void updateMemoryUsage()
             else {
                 userMemoryContext.setBytes(memorySize);
             }
+
+            // This is needed to be fit into UpdateMemory.
```

Comment:
similaraly, we can't we return future from `setBytes` here?

Reply:
Discussion on-going in Slack